### ISSUE TITLE
Does not install with trac 1.6-2~bpo12+1 (debian stable package)

### ISSUE THREAD

**Issue by dimonic** [trac-hacks/trac-subtickets-plugin#37]
Using the current stable trac for debian, none of the installation methods results in a working (or detected) subtickets package. No database upgrade occurs.

**Comment by dimonic** [comment]
Using postgres, this may be something to do with:
`2024-07-24 13:18:01,993 Trac[env] INFO: -------------------------------- environment startup [Trac 1.6] --------------------------------
2024-07-24 13:18:02,211 Trac[loader] ERROR: Skipping "tracsubtickets.api = tracsubtickets.api": ModuleNotFoundError: No module named 'db_default'
2024-07-24 13:18:02,214 Trac[loader] ERROR: Skipping "tracsubtickets.web_ui = tracsubtickets.web_ui": ImportError: cannot import name 'ITemplateStreamFilter' from 'trac.web.api' (/usr/lib/python3/dist-packages/trac/web/api.py)
`

**Comment by hasienda** [comment]
Right, no surprise since ITemplateStreamFilter is not existing anymore in Trac 1.6.

Generic informationen is available for replacing that https://trac.edgewall.org/wiki/TracDev/PortingFromGenshiToJinja#ReplacingITemplateStreamFilter methods by equal functionality. It means accounting for the switch from Genshi to Jinja2 in serverside HTML generation. Details depend a lot on plugin specific ways and extent of native Trac page modifications for display of and interaction with plugin-specific content.

**Comment by dimonic** [comment]
I managed to fix the core issue (db_default import) by adding "from ." to comply with python3 directory stuff. I could submit that as a git commit.

**Comment by snev52** [comment]
I am also interested in a solution. Can you please post details on the fix or submit it? Thank you.

Edit: The code from pull-request #34 worked for me with trac 1.6 on Ubuntu 24.04 with subversion / postgres. (#36 had issues, I think with postgres database)

**Comment by jun66j5** [comment]
I've merged #36 with cosmetic changes to make compatible with Trac 1.6 and Python 3. Feel free to reopen this issue if you encounter any issues while trying master branch with Trac 1.6.
